Zend/zend_ast.c

```diff
@@ -54,6 +54,18 @@ ZEND_API zend_ast * ZEND_FASTCALL zend_ast_create_znode(znode *node) {
 	return (zend_ast *) ast;
 }
 
+ZEND_API zend_ast * ZEND_FASTCALL zend_ast_create_fcc(void) {
+	zend_ast_fcc *ast;
+
+	ast = zend_ast_alloc(sizeof(zend_ast_fcc));
+	ast->kind = ZEND_AST_CALLABLE_CONVERT;
+	ast->attr = 0;
+	ast->lineno = ZEND_MAP_PTR_NEW_OFFSET();
```

Comment:
Optimally we would only allocate an offset in constant expressions. This could be achieved by setting a global to indicate if we're in a constant expression in the parser. It's not the cleanest solution, but there are other globals we set this way.

Reply:
Or, obviously much easier, just moving this to the compiler.

Reply:
See https://github.com/php/php-src/pull/17213#discussion_r1946248972. With the lineno being fixed to actually be the lineno, the `fptr` field will already only be initialized in `zend_compile_const_expr_fcc()`.